hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/token/HapiTokenCreate.java

```diff
@@ -329,7 +329,16 @@ protected Consumer<TransactionBody.Builder> opBodyDef(final HapiSpec spec) throw
                             supplyType.ifPresent(b::setSupplyType);
                             symbol.ifPresent(b::setSymbol);
                             name.ifPresent(b::setName);
-                            entityMemo.ifPresent(s -> b.setMemo(s));
+                            // If the token doesn't have an explicit memo (which
+                            // should help us correlate any related failures to the
+                            // originating spec), then we set the spec name as the
+                            // memo for this purpose
+                            entityMemo.ifPresentOrElse(
```

Comment:
Great feature, I bet this'll help

Reply:
It may be helpful to distinguish auto-set memos with a prefix of some kind. Just a thought